Realm/Realm/DataBinding/WovenSetterMethodInfo.cs

```diff
@@ -60,23 +60,19 @@ public WovenSetterMethodInfo(MethodInfo mi)
         public override object Invoke(object obj, BindingFlags invokeAttr, Binder binder, object[] parameters, CultureInfo culture)
         {
             var managingRealm = (obj as RealmObjectBase)?.Realm;
-            Transaction writeTransaction = null;
-            if (managingRealm != null && !managingRealm.IsInTransaction)
-            {
-                writeTransaction = managingRealm.BeginWrite();
-            }
 
-            var result = _mi.Invoke(obj, invokeAttr, binder, parameters, culture);
-
-            if (writeTransaction != null)
+            // If managingRealm is null or is currently in transaction, just set the property
+            if (managingRealm?.IsInTransaction == false)
             {
-                writeTransaction.Commit();
-                writeTransaction.Dispose();
```

Comment:
There was a small risk that something before the dispose would throw, thus never reaching the actual dispose call.